Compatibility with OpenSearch 2.12.0

## Description
We need to ensure the compatibility with the next version of OpenSearch v2.12.0.
This update is still being discussed, but we need to be aware of potential issues.

For that, we need to:

- [x] Review opensearch's release notes. 
- [x] Identify improvements and potential impact.
- [x] Create a new development branch.
- [x] Develop a testing environment to verify our components would work under this new build.
- [x] Update repository environment variable to 2.12.0 (required by GH Actions).

## Issues
-  _List here the detected issues_

Docker environment fails with:

```
> Task :run FAILED
Exec output and error:
| Output for ./bin/opensearch-keystore:env: can't execute 'bash': No such file or directory

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command './bin/opensearch-keystore'' finished with non-zero exit value 127

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.

BUILD FAILED in 4s
```

--- a/docker/dev/images/Dockerfile
+++ b/docker/dev/images/Dockerfile
@@ -7,11 +7,12 @@ RUN gradle clean
 
 FROM eclipse-temurin:17-jdk-alpine
 RUN apk add git && \
-		apk add curl && \
-    addgroup -g 1000 wazuh-indexer && \
-    adduser -u 1000 -G wazuh-indexer -D -h /home/wazuh-indexer wazuh-indexer && \
-    chmod 0775 /home/wazuh-indexer && \
-    chown -R 1000:0 /home/wazuh-indexer
+  apk add curl && \
+  apk add bash && \
+  addgroup -g 1000 wazuh-indexer && \
+  adduser -u 1000 -G wazuh-indexer -D -h /home/wazuh-indexer wazuh-indexer && \
+  chmod 0775 /home/wazuh-indexer && \
+  chown -R 1000:0 /home/wazuh-indexer
 USER wazuh-indexer
 COPY --from=builder --chown=1000:0 /home/wazuh-indexer /home/wazuh-indexer
 WORKDIR /home/wazuh-indexer

